tests/unit/test_faultydata.py

```
@@ -0,0 +1,101 @@
+
+    def testInstantiation(self):
+        for variantSetId in self.setIds:
+            path = self.getFullPath(variantSetId)
+            self.assertRaises(
+                exceptions.NotIndexedException,
+                variants.HtslibVariantSet, variantSetId, path)
+
+
+class TestInconsistentMetaData(FaultyDataTest):
+    setIds = ["inconsist_meta"]
+
+    def testInstantiation(self):
+        for variantSetId in self.setIds:
+            path = self.getFullPath(variantSetId)
+            self.assertRaises(
+                exceptions.InconsistentMetaDataException,
+                variants.HtslibVariantSet, variantSetId, path)
+
+
+class TestInconsistentCallSetId(FaultyDataTest):
+    setIds = ["inconsist_sampleid", "inconsist_sampleid2"]
+
+    def testInstantiation(self):
+        for variantSetId in self.setIds:
+            path = self.getFullPath(variantSetId)
+            self.assertRaises(
+                exceptions.InconsistentCallSetIdException,
+                variants.HtslibVariantSet, variantSetId, path)
+
+
+class TestOverlappingVariants(FaultyDataTest):
+    setIds = ["overlapping_vcf"]
+
+    def testInstantiation(self):
+        for variantSetId in self.setIds:
+            path = self.getFullPath(variantSetId)
+            self.assertRaises(
+                exceptions.OverlappingException,
+                variants.HtslibVariantSet, variantSetId, path)
+
+
+class TestEmptyDirException(FaultyDataTest):
+    setIds = ["empty_dir"]
+
+    def testInstantiation(self):
+        for variantSetId in self.setIds:
+            path = self.getFullPath(variantSetId)
+            self.assertRaises(
+                exceptions.EmptyDirException,
+                variants.HtslibVariantSet, variantSetId, path)
+
+
+class TestDuplicateCallSetId(FaultyDataTest):
+    """
+    THIS SECTION IS CURRENTLY NOT WORKING
+    It returns the following error:
+
+    [E::bcf_hdr_add_sample] Duplicated sample name 'S1'
+    Aborted (core dumped)
```

It is an htslib bug where adding a duplicate sample id triggers an `abort()` if verbosity >= 2.  There are quite a few other places where `abort()` is called instead of returning an error code.

See htslib/vcf.c lines 79-83:

        if (hts_verbose >= 2)
        {
            fprintf(stderr, "[E::%s] Duplicated sample name '%s'\n", __func__, s);
            abort();
        }